### PR TITLE
Migrate Facebook CTL tests to Playwright

### PR DESCRIPTION
**Pull request by sammacbeth**
**Reviewer:**

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: integration-test/click-to-load-facebook.spec.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ * @param {LoggedRequestDetails[]} requests
+ *   Array of request details, appended to as requests happen.
+ *   Note: The requests array is mutated by this function.
+ * @param {function} [filter]
+ *   Optional filter function that (if given) should return falsey for requests
+ *   that should be ignored.
+ * @returns {Promise<function>}
+ *   Function that clears logged requests (and in progress requests).
+ */
+async function logPageRequests (page, requests, filter) {
```
Comment (kzar) [inline integration-test/click-to-load-facebook.spec.js:23]: Perhaps put this in `helpers/requests.js`? In fact, this looks very similar to the existing helper there, so perhaps you could expand the existing helper to support Playwright?

File: integration-test/click-to-load-facebook.spec.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        // When the page is reloaded, requests should be blocked again.
+        clearRequests()
+        await page.reload({ waitUntil: 'networkidle', timeout: 15000 })
+        {
+            const {
+                requestCount, blockCount, allowCount, facebookSDKRedirect
+            } = summariseFacebookRequests(pageRequests)
+
+            expect(facebookSDKRedirect.checked).toBe(true)
+            // FIXME - It seems that requests to the SDK are not reliably
```
Comment (kzar) [inline integration-test/click-to-load-facebook.spec.js:218]: Nit: Can this comment be removed now?

File: integration-test/helpers/requests.js [hunk 3]
```diff
@@ -29,17 +25,15 @@
  * @returns {Promise<function>}
  *   Function that clears logged requests (and in progress requests).
  */
-async function logPageRequests (page, requests, filter) {
+async function logPageRequests (page, requests, filter, isPlaywright = false) {
```
Comment (kzar) [inline integration-test/helpers/requests.js:28]: Could we set `isPlaywright` inside the function by checking if `page instanceof Puppeteer.Page` or similar? Then we could avoid needing the optional parameter.

Comment (sammacbeth) [reply integration-test/helpers/requests.js]: I was trying to find a good way to do that, your suggestion work so I'm using that now.

**Review by kzar — APPROVED** [review]
LGTM

**Pull request merged**